lib/card.rb

```diff
@@ -14,11 +14,9 @@ class Card < ActiveRecord::Base
 
   has_many :references_from, :class_name => :Reference, :foreign_key => :referee_id
   has_many :references_to,   :class_name => :Reference, :foreign_key => :referer_id
-  has_many :acts, :order => :id
-  has_many :actions, :order => :id, :conditions=>{:draft => [nil,false]}
-  has_many :drafts, :order=>:id, :conditions=>{:draft=>true}, :class_name=> :Action
-
-  cache_attributes 'name', 'type_id' # review - still worth it in Rails 3?
+  has_many :acts, -> { order :id }
+  has_many :actions, -> { where( :draft=>[nil,false]).order :id }
+  has_many :drafts, { :class_name=> :Action }, -> { where( :draft=>true ).order :id }
```

Comment:
I don't know what you meant about BOTH 3.2.x and 4.2.x version.  The initial idea was to stop supporting 3.2.16 (which was the only version we supported before; not 3.2.15 or 3.2.17) and start using ~> 4.2 (which is different from 4.2.x because it includes 4.3 but not 4.1).

It took me a while to figure out what "platform use case" meant.  Could we refer to this as "wagn-as-rails-engine" or "decko-rails"?  If anything, the "platform use case" would suggest the old way, not the new (let's call the old way "wagn-as-platform" or maybe "decko-based").

At any rate, you're point is well taken.  Whereas in our primary use case (decko-based) we we're fine to specify the rails version, in our new use case (decko-rails) we need to support a decent range of versions or we're going to cause problems.

As for where to draw the line, it would be a lot simpler to support 4.x rather than 3.2.x and 4.x.  And since Rails 4 has been out for a year and a half now, we'd only need 3.2.x to support adding decko-rails to fairly old applications.  But presumably this is *your* use case, which makes it an important one.  Exactly what version are you using now?

Here's what I would propose: let's keep moving this branch forward with a focus on figuring out what has to happen to get things to work in the latest rails code (4.2).  We've already got it working in 3.2.16 (the current code).  Once we have those both worked out, we can better assess how hard it would be to go back as far as 3.2.x.  If we're just talking about a dozen or so "if" statements to keep 3.2 going, it's probably worth it, at least until 2.0.  I'm not at all excited about limiting ourselves to 3.2 functionality much past Decko 1.0.

I'm really itching to start requiring 1.9, too.  That was released many years ago now, so it feels ancient not to be moving forward on that!